src/app/pages/storage/modules/pool-manager/components/pool-manager-wizard/components/pool-warnings/pool-warnings.component.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { ReactiveFormsModule, FormGroup, FormControl } from '@angular/forms';
+import { MatCheckboxHarness } from '@angular/material/checkbox/testing';
+import { Spectator } from '@ngneat/spectator';
+import { mockProvider, createComponentFactory } from '@ngneat/spectator/jest';
+import { of } from 'rxjs';
+import { UnusedDisk } from 'app/interfaces/storage.interface';
+import { IxRadioGroupHarness } from 'app/modules/ix-forms/components/ix-radio-group/ix-radio-group.harness';
+import { IxFormsModule } from 'app/modules/ix-forms/ix-forms.module';
+import { PoolWarningsComponent } from 'app/pages/storage/modules/pool-manager/components/pool-manager-wizard/components/pool-warnings/pool-warnings.component';
+import { GeneralWizardStepComponent } from 'app/pages/storage/modules/pool-manager/components/pool-manager-wizard/steps/1-general-wizard-step/general-wizard-step.component';
+import { PoolManagerStore } from 'app/pages/storage/modules/pool-manager/store/pool-manager.store';
+
+const duplicateSerialDisk = {
+  identifier: '{uuid}bb73faf5-6d50-4af9-ae30-bc8ba0cf8694',
+  duplicate_serial: ['sdb', 'sdc'],
+  devname: 'sda',
+} as UnusedDisk;
+
+const exportedPoolDisk = {
+  identifier: '{uuid}bb73faf5-6d50-4af9-ae30-bc8ba0cf8695',
+  duplicate_serial: [],
+  exported_zpool: 'FAKE_POOL',
+  devname: 'sdb',
+} as unknown as UnusedDisk;
+
+const duplicateSerialAndExportedPoolDisk = {
+  identifier: '{uuid}bb73faf5-6d50-4af9-ae30-bc8ba0cf866',
+  duplicate_serial: ['sdb'],
+  exported_zpool: 'MOCK_POOL',
+  devname: 'sdc',
+} as unknown as UnusedDisk;
+
+describe('PoolWarningsComponent', () => {
+  let spectator: Spectator<PoolWarningsComponent>;
+  let loader: HarnessLoader;
+  let form: GeneralWizardStepComponent['form'];
+
+  const createComponent = createComponentFactory({
+    component: PoolWarningsComponent,
+    imports: [
+      ReactiveFormsModule,
+      IxFormsModule,
+    ],
+    providers: [
+      mockProvider(PoolManagerStore, {
+        allDisks$: of([duplicateSerialDisk, exportedPoolDisk, duplicateSerialAndExportedPoolDisk]),
+        setDiskWarningOptions: jest.fn(),
+      }),
+    ],
+  });
+
+  beforeEach(() => {
+    form = new FormGroup({
```

Comment:
Once you move form inside of the component, it's better to check whether correct method in store was called instead of checking form values.